Rename scenarios to scripts (#8147)
digital-asset/daml: docs/source/daml/intro/5_Restrictions.rst

For details, see :ref:`Background concepts - time <time>`.

> Time in scripts

Author: ```suggestion
Time in test scripts
```

~~~~~~~~~~~~~~~~~

> In scripts, record and ledger time are always equal. You can set them using the following functions:

Reviewer: I guess this is not true anymore for a script running against a real ledger?

Author: Oh, good catch, I was just blindly replacing. How about this?
```suggestion
In test scripts, record and ledger time are always equal. You can set them using the following functions:
```

Each has a special action ``abort txt`` that represents failure, and that takes on type ``Update ()`` or ``Script ()`` depending on context .

> Transactions and scripts succeed or fail *atomically* as a whole. So an occurrence of an ``abort`` action will always fail the **entire** evaluation of the current ``Script`` or ``Update``.

Reviewer 2: Scripts are not atomic, neither are scenarios. Each `submit` is atomic but if you have multiple submits and it fails after the first few, those will still have been applied to the ledger. Same behavior as for any other ledger.

Author: Understood, good catch again! How about this?
```suggestion
Transactions and scripts can either successfully return or fail. So an occurrence of an ``abort`` action will always fail the **entire** evaluation of the current ``Script`` or ``Update``.
```

Reviewer 2: I’d say atomicity of transactions is pretty crucial so I would keep the `atomic` part but remove the mention of scripts. The fact that a script can fail or succeed doesn’t really seem like useful information.

Author: Well this subsection has the title "Failing actions", so I guess it could be useful to mention that a script can fail.
How about this?
```suggestion
Transactions succeed or fail *atomically* as a whole. So an occurrence of an ``abort`` action will always fail the **entire** evaluation of the current ``Update``. Naturally, a script with a failed transaction will fail as well. 
```

Reviewer 2: Sounds reasonable. Technically I guess you can bikeshed and if the failed transaction is in `submitMustFail` the script will succeed but I don’t think that level of detail is helpful here.

Author: Yeah, I also hesitated a bit adding a note like "There is one exception: a failing transaction wrapped in ``submitMustFail`` will not cause the script to fail."
If it makes any reader wonder, maybe we should add it?

Reviewer 2: How about the following: “Transactions succeed or fail atomically as a whole. Scripts on the other hand do not fail atomically: While each `submit` is atomic, if one submit succeeded and the script fails later, the effects of that submit will still be applied to the ledger”.

Author: I like it. I formatted it, made While lowercase, s/one/a/, if you agree, please commit this change to resolve this thread:
```suggestion
Transactions succeed or fail *atomically* as a whole. Scripts on the other hand do not fail atomically: while each ``submit`` is atomic, if a ``submit`` succeeded and the script fails later, the effects of that ``submit`` will still be applied to the ledger.
```
